torch/nested/_internal/nested_tensor.py

```diff
@@ -163,6 +163,19 @@ def offsets(self):
     def lengths(self):
         return self._lengths
 
+    def share_memory_(self):
```

Comment:
instead of implementing the method directly on the subclass, it's better to override it via NJT's `__torch_function__()` implementation here: https://github.com/pytorch/pytorch/blob/f638854e1da6b33e78dcc9f3e28c98c4cdce4e86/torch/nested/_internal/ops.py#L370

this should compose better with the rest of PyTorch.